CHANGELOG.rst

```diff
@@ -0,0 +1,43 @@
+=========
```

Comment:
I choose the rst format because the documentation is written in this format, therefore in the future it should be easier to include the changelog into the documentation.

Check it how it looks like [on github](https://github.com/memgraph/pymgclient/blob/T0411-MG-fix-memory-leak/CHANGELOG.rst).